Content.Server/Humanoid/Systems/HumanoidSystem.cs

```diff
@@ -94,11 +94,13 @@ public void LoadProfile(EntityUid uid, HumanoidCharacterProfile profile, Humanoi
 
         SetSkinColor(uid, profile.Appearance.SkinColor, false);
         SetBaseLayerColor(uid, HumanoidVisualLayers.Eyes, profile.Appearance.EyeColor, false);
+        humanoid.CachedEyeColor = profile.Appearance.EyeColor;
 
         humanoid.CurrentMarkings.Clear();
 
         // Hair/facial hair - this may eventually be deprecated.
 
+        humanoid.CachedHairColor = profile.Appearance.HairColor;
```

Comment:
one of the current issues with this is that markings as hair was supposed to imply that there's a variable number of hair types and colors for a single character: could you find a way to, instead of using the hair color field in the appearance object, use a single hair marking's color as the 'base hair color' for this value? as the comment just above says, the hair/facial hair fields were meant to be deprecated at some point in time, but kept as a legacy codepath so that an opinion could be formed on keeping those fields or not

Reply:
Now it loops through all markings for hair when it have to be defined. It also can't be defined if you don't have hair. However in UI hair colors not used from markings but from hair color slider. So there can be issues with that when selected species can't have hair or bald.

Reply:
hair style ID and color was intended to be deprecated and stored in the marking set - you can cache a hair color after the marking set is populated, however, by just using `TryGetCategory` using the Hair category enum variant on the humanoid component's marking set, seeing if the category is populated or empty, and then just getting the first hair from that list and then using the first color on that marking definition or some default value otherwise

Reply:
```cs
foreach (var marking in profile.Appearance.Markings)
        {
            // Caching hair and facial hair colors from their markings
            if (_markingManager.Markings.TryGetValue(marking.MarkingId, out var prototype)){
                switch (prototype.MarkingCategory) {
                    case MarkingCategories.Hair:
                        humanoid.CachedHairColor = marking.MarkingColors[1];
                        break;
                    case MarkingCategories.FacialHair:
                        humanoid.CachedFacialHairColor = marking.MarkingColors[1];
                        break;
                }
            }
            
            AddMarking(uid, marking.MarkingId, marking.MarkingColors, false);
        }
```
isn't it good solution? I just added some checks in loop

This is how it looks in master branch
```cs
        // Hair/facial hair - this may eventually be deprecated.

        AddMarking(uid, profile.Appearance.HairStyleId, profile.Appearance.HairColor, false);
        AddMarking(uid, profile.Appearance.FacialHairStyleId, profile.Appearance.FacialHairColor, false);

        foreach (var marking in profile.Appearance.Markings)
        {
            AddMarking(uid, marking.MarkingId, marking.MarkingColors, false);
        }
```